www/include/home/customViews/action.php

```diff
@@ -46,105 +47,293 @@
 }
 
+$action = $_POST['action'];
+
+$postFilter = array(
+    'widget_id' => [
+        'filter' => FILTER_VALIDATE_INT,
+        'options' => [
+            'default' => false
+        ]
+    ],
+    'custom_view_id' => [
+        'filter' => FILTER_VALIDATE_INT,
+        'options' => [
+            'default' => false
+        ]
+    ],
+    'timer' => [
+        'filter' => FILTER_VALIDATE_INT,
+        'options' => [
+            'default' => false
+        ]
+    ],
+    'public' => [
+        'filter' => FILTER_VALIDATE_INT,
+        'options' => [
+            'default' => false
+        ]
+    ],
+    'user_id' => [
+        'filter' => FILTER_VALIDATE_INT,
+        'options' => [
+            'default' => false
+        ]
+    ],
+    'viewLoad' => [
+        'filter' => FILTER_VALIDATE_INT,
+        'options' => [
+            'default' => false
+        ]
+    ],
+    'name' => [
+        'filter' => FILTER_SANITIZE_STRING,
+        'options' => [
+            'default' => ''
+        ]
+    ],
+    'layout' => [
+        'options' => [
+            'default' => ''
+        ]
+    ],
+    'widget_model_id' => [
+        'filter' => FILTER_VALIDATE_INT,
+        'options' => [
+            'default' => false
+        ]
+    ],
+    'widget_model_id' => [
+        'filter' => FILTER_VALIDATE_INT,
+        'options' => [
+            'default' => false
+        ]
+    ],
+    'widget_title' => [
+        'filter' => FILTER_SANITIZE_STRING,
+        'options' => [
+            'default' => ''
+        ]
+    ]
+);
+
+$postInputs = filter_input_array(INPUT_POST, $postFilter);
+
+$lockedUsers = [];
+if (!empty($_POST['lockedUsers'])) {
+    foreach ($_POST['lockedUsers'] as $lockedUserId) {
+        if (filter_var($lockedUserId, FILTER_VALIDATE_INT) !== false) {
+            $lockedUsers[] = (int) $lockedUserId;
+        }
+    }
+}
+$unlockedUsers = [];
+if (!empty($_POST['unlockedUsers'])) {
+    foreach ($_POST['unlockedUsers'] as $unlockedUserId) {
+        if (filter_var($unlockedUserId, FILTER_VALIDATE_INT) !== false) {
+            $unlockedUsers[] = (int) $unlockedUserId;
+        }
+    }
+}
+$lockedUsergroups = [];
+if (!empty($_POST['lockedUsergroups'])) {
+    foreach ($_POST['lockedUsergroups'] as $lockedUsergroupsId) {
+        if (filter_var($lockedUsergroupsId, FILTER_VALIDATE_INT) !== false) {
+            $lockedUsergroups[] = (int) $lockedUsergroupsId;
+        }
+    }
+}
+$unlockedUsergroups = [];
+if (!empty($_POST['unlockedUsergroups'])) {
+    foreach ($_POST['unlockedUsergroups'] as $unlockedUsergroupsId) {
+        if (filter_var($unlockedUsergroupsId, FILTER_VALIDATE_INT) !== false) {
+            $unlockedUsergroups[] = (int) $unlockedUsergroupsId;
+        }
+    }
+}
+
+$positions = [];
+if (!empty($_POST['positions'])) {
+    foreach ($_POST['positions'] as $position) {
+        if (filter_var($position, FILTER_SANITIZE_STRING) !== false) {
+            $positions[] = $position;
+        }
+    }
+}
```

Comment:
as @kduret noticed, you don't need to check if array is empty, it will already avoid loop if empty

Reply:
The $_POST is defined but not the `$_POST['index'].` So it will enter the loop and check for the index. Since those values are coming from a multiples select2 it is only set when user selects something.

The issue here is not that the array is empty, but the array is not defined.

So we need to keep this check